### PR TITLE
Closes 1462 Increase test coverage for derive_param_tte

### PR DESCRIPTION
**Pull request by jeffreyad**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately 
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: tests/testthat/test-derive_param_tte.R [hunk 1]
```diff
@@ -633,3 +633,14 @@ test_that("derive_param_tte Test 9: error is issued if parameter code already ex
...
+
+## Test 10: error is issued if package does not exist ----
+test_that("derive_param_tte Test 10: error is issued if package does not exist", {
+
+  expected_output <- list_tte_source_objects(package = "admiral")
+
+  expect_error(
+    list_tte_source_objects(package = "tte"),
+    regexp = "No package called 'tte' is installed and hence no `tte_source` objects are available"
+  )
```
Comment (thomas-neitmann) [inline tests/testthat/test-derive_param_tte.R:645]: If you check for a warning or error there's no need for an "expected" output apart from the `regexp` you define inside `expect_error()`.

```suggestion
  expect_error(
    list_tte_source_objects(package = "tte"),
    regexp = "No package called 'tte' is installed and hence no `tte_source` objects are available"
  )
```

File: tests/testthat/test-derive_param_tte.R [hunk 2]
```diff
@@ -633,3 +633,14 @@ test_that("derive_param_tte Test 9: error is issued if parameter code already ex
     regexp = "^The parameter code 'TTAE' does already exist in `dataset`.$"
   )
 })
+
+## Test 10: error is issued if package does not exist ----
+test_that("derive_param_tte Test 10: error is issued if package does not exist", {
```
Comment (thomas-neitmann) [inline tests/testthat/test-derive_param_tte.R:638]:
```suggestion
test_that("list_tte_source_objects Test 1: error is issued if package does not exist", {
```

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

**Comment by jeffreyad** [comment]
@thomas-neitmann added test 11 to cover code under # Get all `tte_source` objects exported by `package`.  Let me know if there is a better way to do this.  Thanks

image

**Comment by thomas-neitmann** [comment]
> @thomas-neitmann added test 11 to cover code under # Get all tte_source objects exported by package. Let me know if there is a better way to do this. Thanks

RIght now you're just checking whether the code runs without errors in test 11. What you'd rather want to do is define an `expected_output` and compare that against whatever `list_tte_source_objects()` returns.

File: tests/testthat/test-derive_param_tte.R [hunk 3]
```diff
@@ -630,3 +630,39 @@ test_that("derive_param_tte Test 9: error is issued if parameter code already ex
     regexp = "^The parameter code 'TTAE' does already exist in `dataset`.$"
   )
 })
+
+## Test 10: error is issued if package does not exist ----
+test_that("list_tte_source_objects Test 1: error is issued if package does not exist", {
```
Comment (bms63) [inline tests/testthat/test-derive_param_tte.R:635]:
```suggestion
test_that("list_tte_source_objects Test 10: error is issued if package does not exist", {
```

File: tests/testthat/test-derive_param_tte.R [hunk 4]
```diff
@@ -630,3 +630,39 @@ test_that("derive_param_tte Test 9: error is issued if parameter code already ex
...
+  expect_error(
+    list_tte_source_objects(package = "tte"),
+    regexp = "No package called 'tte' is installed and hence no `tte_source` objects are available"
+  )
+})
+
+## Test 11: calling list_tte_source_objects results in expected output objects ----
+test_that("list_tte_source_objects Test 2: expected objects produced", {
```
Comment (bms63) [inline tests/testthat/test-derive_param_tte.R:643]:
```suggestion
test_that("list_tte_source_objects Test 11: expected objects produced", {
```

**Review by bms63 — CHANGES_REQUESTED** [review]
couple of typos

**Comment by bms63** [comment]
@jeffreyad omg Jeff I thought I was going crazy!!!  I was like why did it auto-update...lol  probably should work on the weekend!!

**Review by bms63 — APPROVED** [review]
LGTM after the checks pass

**Pull request merged**